### PR TITLE
Support keys with dots and annotations in README

### LINKED ISSUE #22: undefined value for parameters with .dot in the key
Version: readme-generator-for-helm@1.3.0

Example:

```yaml
## @section Testing Parameters
## @param this.works a param that works
## @param t.testing.me nested param with dot not working
this.works: "fine"
t:
  testing.me: 123
```

Renders:

### Testing Parameters

| Name           | Description                       | Value       |
| -------------- | --------------------------------- | ----------- |
| `this.works`   | a param that works                | `fine`      |
| `t.testing.me` | nested param with dot not working | `undefined` |

---
Parameters with dot in their key are not the smartest idea, but it is out their, real life example:

```yaml
serviceAccount:
  annotations:
    iam.gke.io/gcp-service-account: <accountname>@<GCP-project-name>.iam.gserviceaccount.com
```

### PR DESCRIPTION
**Pull request by miguelaeh**
Signed-off-by: Miguel A. Cabrera Minagorri <mcabrera@vmware.com>

**Description of the change**

This PR adds support for common K8s annotations and keys using dots and/or other characters combined.

It supports the rendering of that kind of parameter in the READMEs, but not in the schemas. This is because the schema generation needs some refactorization to support it.

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #22 

**Additional information**

**Checklist**

- [x] Version bumped in `package.json` and `package-lock.json` according to [semver](http://semver.org/).
- [x] New features are documented in the README.md
- [x] New features contain a new test at the `tests` folder
- [x] All tests pass running `npm test`

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -39,8 +39,31 @@ function containsModifier(parameter, modifier) {
...
+function getArrayPath(obj, path) {
+  let fullPath = [];
+  console.log(path)
+  for (let key of Object.keys(obj)) {
+    if (path === key) {
+      // We are the tail of the tree
+      fullPath.push(key);
+    } else {
+      const subKeys=path.split(`${key}.`);
+      if (subKeys.length > 1 && path.startsWith(`${key}.`)) {
```
Comment (FraPazGal) [inline lib/utils.js:54]: Not sure if I'm understanding this right. What would be the behavior of this if, for example, we encountered the following corner case:
```
  ingress:
    hostname:
      - "hostname.test"
```

File: lib/builder.js [hunk 2]
```diff
@@ -69,6 +69,8 @@ function combineMetadataAndValues(valuesObject, valuesMetadata) {
         // Set the value from actual object
         param.value = valuesObject[paramIndex].value;
         param.type = valuesObject[paramIndex].type;
+        // TODO(miguelaeh): Hack to avoid render parameters with dots in keys into the schema. Must be removed once fixed
+        param.schema = valuesObject[paramIndex].schema;
```
Comment (FraPazGal) [inline lib/builder.js:73]: I understand we are taking the value from the one defined in [parser.js](https://github.com/bitnami-labs/readme-generator-for-helm/pull/43/files#diff-2f45270c7389933f71b380538ff52e106ee5cb66a0b573d9967467dfdbc11c02) but I'm missing how the render is avoiding this parameter so it is not added to the schema.

Comment (FraPazGal) [inline lib/builder.js:73]: I understand we are taking the value from the one defined in [parser.js](https://github.com/bitnami-labs/readme-generator-for-helm/pull/43/files#diff-2f45270c7389933f71b380538ff52e106ee5cb66a0b573d9967467dfdbc11c02) but I'm missing how the render is avoiding this parameter so it is not added to the schema.

**Review by FraPazGal — COMMENTED** [review]
The PR LGTM overall, though I have a couple of minor questions. Please look at them when possible.

File: lib/utils.js (hunk 1 above)
Comment (miguelaeh) [reply lib/utils.js:54]: that should not be a corner case as what has a dot there is the value not the key. Could you clarify the question?

File: lib/builder.js (hunk 2 above)
Comment (miguelaeh) [reply lib/builder.js:73]: Because by default, all parameters are rendered into the schema. Setting the property to the value of `valuesObject[paramIndex].schema`, it will ship a boolean from the parse step in which there is a similar hack setting it to false until we refactor the schema-generation

File: lib/utils.js (hunk 1 above)
Comment (FraPazGal) [reply lib/utils.js:54]: You are right, I double-checked this and it would work as expected.

File: lib/builder.js (hunk 2 above)
Comment (FraPazGal) [reply lib/builder.js:73]: Thanks for the explanation, LGTM!

**Review by FraPazGal — APPROVED** [review]
